src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -128,13 +134,17 @@ public Http2Connection(HttpConnectionPool pool, Stream stream)
 
             _httpStreams = new Dictionary<int, Http2Stream>();
 
-            _connectionWindow = new CreditManager(this, nameof(_connectionWindow), DefaultInitialWindowSize);
+            _connectionWindow = new CreditManager(this, nameof(_connectionWindow), DefaultInitialConnectionWindowSize);
             _concurrentStreams = new CreditManager(this, nameof(_concurrentStreams), InitialMaxConcurrentStreams);
+            InitialStreamWindowSize = pool.Settings._initialStreamWindowSize;
+            _rttEstimator = _pool.Settings._fakeRtt != null || socket != null ?
+                new RttEstimator(this, _pool.Settings._fakeRtt, socket) :
+                null;
 
             _writeChannel = Channel.CreateUnbounded<WriteQueueEntry>(s_channelOptions);
 
             _nextStream = 1;
-            _initialWindowSize = DefaultInitialWindowSize;
+            _initialWindowSize = DefaultInitialConnectionWindowSize;
```

Comment:
This is weird now that you've change the name of the constant to DefaultInitial*Connection*WindowSize.

_initialWindowSize tracks the current initial window size for *streams*.

I think we should either leave the naming as-is, or define two separate constants, one for connection and one for stream. I would suggest `InitialConnectionWindowSize` and `DefaultInitalStreamWindowSize`. These would both have the same value (65535) but perhaps having two different constants makes their usage clearer.